### PR TITLE
stats: correct response_code regex

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fixes: #39903 
wait for ~~https://github.com/envoyproxy/envoy/pull/22791~~ https://github.com/envoyproxy/envoy/pull/23024

support multiple regexes to populate the same tag name.

before:

```
envoy_cluster_internal_upstream_rq_200{cluster_name="xds-grpc"} 1
envoy_cluster_internal_upstream_rq{response_code_class="2xx",cluster_name="xds-grpc"} 1
envoy_cluster_upstream_rq{response_code_class="2xx",cluster_name="xds-grpc"} 1
istio_requests_total{response_code="200",reporter="source",source_workload="sleep",source_workload_namespace="default",source_principal="spiffe://cluster.local/ns/default/sa/sleep",source_app="sleep",source_version="unknown",source_cluster="Kubernetes",destination_workload="httpbin",destination_workload_namespace="default",destination_principal="spiffe://cluster.local/ns/default/sa/httpbin",destination_app="httpbin",destination_version="v1",destination_service="httpbin.default.svc.cluster.local",destination_service_name="httpbin",destination_service_namespace="default",destination_cluster="Kubernetes",request_protocol="http",response_flags="-",grpc_response_status="",connection_security_policy="unknown",source_canonical_service="sleep",destination_canonical_service="httpbin",source_canonical_revision="latest",destination_canonical_revision="v1"} 1
```

after:

```
envoy_cluster_internal_upstream_rq{response_code="200",cluster_name="xds-grpc"} 1
envoy_cluster_internal_upstream_rq{response_code_class="2xx",cluster_name="xds-grpc"} 1
envoy_cluster_upstream_rq{response_code="200",cluster_name="xds-grpc"} 1
envoy_cluster_upstream_rq{response_code_class="2xx",cluster_name="xds-grpc"} 1
istio_requests_total{response_code="200",reporter="source",source_workload="sleep",source_workload_namespace="default",source_principal="spiffe://cluster.local/ns/default/sa/sleep",source_app="sleep",source_version="unknown",source_cluster="Kubernetes",destination_workload="httpbin",destination_workload_namespace="default",destination_principal="spiffe://cluster.local/ns/default/sa/httpbin",destination_app="httpbin",destination_version="v1",destination_service="httpbin.default.svc.cluster.local",destination_service_name="httpbin",destination_service_namespace="default",destination_cluster="Kubernetes",request_protocol="http",response_flags="-",grpc_response_status="",connection_security_policy="unknown",source_canonical_service="sleep",destination_canonical_service="httpbin",source_canonical_revision="latest",destination_canonical_revision="v1"} 1
```

### CODE REVIEW

**Comment by zirain** [comment]
Not stale

**Comment by zirain** [comment]
kindly ping @istio/wg-networking-maintainers 
this PR is ready for review.

**Review by hzxuzhonghu — APPROVED** [review]

File: releasenotes/notes/40729.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: telemetry
+
+issue:
+  - 39903
+
+releaseNotes:
+  - |
+    **Fixed** an issue that allow multiple regexes with same tag name.
```
Comment (ericvn) [inline releasenotes/notes/40729.yaml:10]: Maybe
```suggestion
    **Fixed** an issue that allowed multiple regular expressions with the same tag name.
```

Comment (zirain) [reply releasenotes/notes/40729.yaml]: updated

**Review by ericvn — APPROVED** [review]
Test and RN look good.

**Pull request merged**